website/content/docs/concepts/username-templating.mdx

```diff
@@ -0,0 +1,143 @@
+---
+layout: docs
+page_title: Username Templating
+sidebar_title: Username Templating
+description: >-
+  Username templating are used in some secret engines to allow operators to define
+  how dynamic usernames are generated.
+---
+
+# Username Templating
+
+Some of the secrets engines that generate dynamic users for external systems provide the ability for Vault operators
+to customize how usernames are generated for said external systems. This uses the
+[Go template language](https://golang.org/pkg/text/template/). This page describes the basics of using these templates
+for username generation but does not go into great depth of using the templating language for more advanced usages.
+See the API documentation for the given secret engine to determine if it supports username templating and for more
+details on using it with that engine.
+
+~> When customizing how usernames are generated, take care to ensure you have enough randomness to ensure uniqueness
+   otherwise multiple calls to create the credentials may interfere with each other.
+
+In addition to the functionality built into the Go template language, a number of additional functions are available:
+
+## Available Functions
+
+### String/Character Manipulation
+
+`lowercase` - Lowercases the provided value.<br/>
+**Example**: `{{.FieldName | lowercase}}`
+
+`replace` - Find/replace on the provided value.<br/>
+**Example**: `{{.FieldName | replace - _}}`
+
+`truncate` - truncates the previous value to the provided number of characters.<br/>
+**Example**: `{{.FieldName | truncate 10}}`
+
+`truncate_sha256` - Truncates the previous value to the provided number of characters. The last 8 characters of the
```

Comment:
Should we refer to the value as the "input value" and not "previous value"? I feel that may make more sense. Also `lowercase` and `replace` use "provided value" while `truncate` uses "previous value". Think we should be consistent across the functions either way

Reply:
Good catch. Updated. Let me know how this looks. I moved most of them over to "input" and removed "previous".

Reply:
LGTM